interfaces/builtin/basedeclaration.go

```diff
@@ -34,100 +34,258 @@ series: 16
 revision: 0
 slots:
   bluetooth-control:
+    allow-installation:
+      slot-snap-type:
+        - core
     deny-auto-connection: true
   bluez:
     deny-auto-connection: true
   bool-file:
-    deny-auto-connection: true
+    allow-installation:
+      slot-snap-type:
+        - core
+        - gadget
+    deny-auto-connection: true
+  browser-support:
+    allow-installation:
+      slot-snap-type:
+        - core
   camera:
+    allow-installation:
+      slot-snap-type:
+        - core
     deny-auto-connection: true
   content:
     allow-auto-connection:
       plug-publisher-id:
         - $SLOT_PUBLISHER_ID
+    allow-installation:
+      slot-snap-type:
+        - app
+        - gadget
   cups-control:
+    allow-installation:
+      slot-snap-type:
+        - core
     deny-auto-connection: true
   docker:
+    allow-installation: false
```

Comment:
Without snap declarations, this is going to make the docker snap uninstallable. I think this needs to be this for now:

```
docker:
  deny-auto-connection: true
```